File: src/main/java/xyz/chunkstories/core/CoreContentPlugin.java
//
// This file is a part of the Chunk Stories Core codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.core;

import xyz.chunkstories.api.GameContext;
import xyz.chunkstories.api.client.Client;
import xyz.chunkstories.api.plugin.ChunkStoriesPlugin;
import xyz.chunkstories.api.plugin.PluginInformation;
import xyz.chunkstories.core.cli.FoodCommand;
import xyz.chunkstories.core.logic.EntityLogicListener;
import xyz.chunkstories.core.logic.ItemsLogicListener;

/** 'Glue' for hooking core functions into the base engine */
public class CoreContentPlugin extends ChunkStoriesPlugin {

	private ItemsLogicListener itemsLogic;
	private EntityLogicListener entityLogic;

	//private RenderingEventsListener renderingLogic;

	public CoreContentPlugin(PluginInformation pluginInformation, GameContext pluginExecutionContext) {
		super(pluginInformation, pluginExecutionContext);
		this.getPluginManager().registerCommandHandler("food", new FoodCommand());

		pluginExecutionContext.logger().info("INIT CORE CONTENT PLUGIN");
		if(pluginExecutionContext instanceof Client) {
			pluginExecutionContext.logger().info("Installing additional options");
			((Client)pluginExecutionContext).getConfiguration().addOptions(CoreOptions.INSTANCE.getOptions());
		}
	}

	@Override
	public void onEnable() {
		itemsLogic = new ItemsLogicListener(this);
		pluginExecutionContext.getPluginManager().registerEventListener(itemsLogic, this);

		entityLogic = new EntityLogicListener(this);
		pluginExecutionContext.getPluginManager().registerEventListener(entityLogic, this);

		/*if (this.getPluginExecutionContext() instanceof Client) {
			renderingLogic = new RenderingEventsListener(this, (Client) getPluginExecutionContext());
			pluginExecutionContext.getPluginManager().registerEventListener(renderingLogic, this);
		}*/
	}

	@Override
	public void onDisable() {
		// pluginExecutionContext.getPluginManager().unRegisterEventListener(itemsLogic,
		// this);
	}

}

File: src/main/java/xyz/chunkstories/core/CoreContentPlugin.kt
//
// This file is a part of the Chunk Stories Core codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.core

import xyz.chunkstories.api.client.Client
import xyz.chunkstories.api.plugin.Plugin
import xyz.chunkstories.api.plugin.PluginInformation
import xyz.chunkstories.api.world.GameInstance
import xyz.chunkstories.core.cli.FoodCommand
import xyz.chunkstories.core.logic.EntityLogicListener
import xyz.chunkstories.core.logic.ItemsLogicListener

/** 'Glue' for hooking core functions into the base engine  */
class CoreContentPlugin(pluginInformation: PluginInformation, gameInstance: GameInstance) : Plugin(pluginInformation, gameInstance) {
	private lateinit var itemsLogic: ItemsLogicListener
	private lateinit var entityLogic: EntityLogicListener

	init {
		gameInstance.logger.info("Initializing core content plugin")

		this.gameInstance.pluginManager.registerCommand("food", FoodCommand())

		val engine = gameInstance.engine
		if (engine is Client) {
			gameInstance.logger.info("Registering additional configuration options for the client")
			engine.configuration.addOptions(CoreOptions.options)
		}

		onEnable()
	}

	fun onEnable() {
		/*itemsLogic = ItemsLogicListener(this)
		gameInstance.pluginManager.registerEventListener(itemsLogic, this)

		entityLogic = EntityLogicListener(this)
		gameInstance.pluginManager.registerEventListener(entityLogic, this)*/
	}

	override fun onDisable() { }
}
